### PR TITLE
Autolink mentions for Instagram and Twitter

### PR DESCRIPTION
**Pull request by hengwoon**
We at [Re:amaze](https://www.reamaze.com) are using this to link instagram handles, just thought it might be helpful for others.

- Deprecated `twitter` option
- Added `mention` option with possible values `twitter`, `instagram`
- `mention` option will take precedence over `twitter` option
- (breaking) removed Twitter match and matcher objects. Code that checks `match.getType() == 'twitter'` will need to be changed to check `match.getType() == 'mention' && match.getServiceName() == 'twitter'`

Could use some ideas on how to avoid the last breaking change above.

### CODE REVIEW

**Comment by lunaru** [comment]
+1 would be good to see this merged. It even handles the backward compatible cases of "twitter: true" and "twitter: false" nicely.

**Comment by gregjacobs** [comment]
@hengwoon This is absolutely amazing that you implemented this feature. Thank you so much! 

I am on vacation right now but I'll be back Monday and will check everything out. And I'll give it some thought on how to avoid the breaking change with the `matcher.getType()`, but that's definitely a tough one. Might consider moving to v1.0 for this change. 

Talk soon!

**Comment by gregjacobs** [comment]
Btw, I'm so glad that Autolinker is helping to make what looks to be an awesome product!

**Comment by hengwoon** [comment]
@gregjacobs just wondering if you had some time to review this.

One suggestion I have for keeping backwards compatibility for users that are using a custom replace function would be to have  `Autolinker.match.Mention.getType()` return either `twitter` or `instagram` instead of `mention`. This would however mean that if getType() needs to get changed to `mention` in the future, it would then be breaking change. This also deviates a bit from how multiple subtypes are handled (hashtags).

**Comment by olafleur** [comment]
Seems like a nice addition ! Do you need help to resolve the conflicts of this pull request? If you do, let me know !

**Comment by hengwoon** [comment]
@olafleur I've resolved the merge conflicts and rebased on latest master.

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "autolinker",
-  "version": "0.28.0",
-  "description": "Utility to automatically link the URLs, email addresses, and Twitter handles in a given block of text/HTML",
+  "version": "0.29.0",
```
Comment (olafleur) [inline package.json]: No need to change the version. @gregjacobs will take care of that.

**Comment by lunaru** [comment]
@gregjacobs any thoughts? would be good to see this functionality merged.

**Comment by gregjacobs** [comment]
Finally done! Sorry for the delay.. I decided to do a 1.0 version with the breaking change of `'twitter'`->`'mention'` for `match.getType()`, and removing the `twitter` option altogether. 

Thanks so much for this PR!

**Pull request merged**